tests/unit/test_bs_roformer_fp16.py

```
@@ -0,0 +1,243 @@
+from audio_separator.separator.architectures.mdxc_separator import MDXCSeparator
+from audio_separator.separator.execution_policy import NATIVE_FP16
+from audio_separator.separator.roformer.roformer_loader import RoformerLoader
+from audio_separator.separator.uvr_lib_v5.roformer import bs_roformer as bs_module
+
+
+def _apple_gpu_is_virtualized() -> bool:
+    """Detect a paravirtualized Metal device (hosted CI Macs report VirtualMac*)."""
+    if platform.system() != "Darwin":
+        return False
+    try:
+        model = subprocess.run(["sysctl", "-n", "hw.model"], capture_output=True, text=True, timeout=5, check=False).stdout
```

Ruff (0.16.0)

[error] Starting a process with a partial executable path

(S607)